contracts/exchange/contracts/src/MixinExchangeCore.sol

```diff
@@ -466,9 +466,8 @@ contract MixinExchangeCore is
         internal
         view
     {
-        // Ensure order is valid
-        // An order can only be cancelled if its status is FILLABLE.
-        if (orderInfo.orderStatus != uint8(OrderStatus.FILLABLE)) {
+        // Ensure order has not already been cancelled.
+        if (orderInfo.orderStatus == uint8(OrderStatus.CANCELLED)) {
```

Comment:
I wonder if we should also revert if the order has been filled 🤔. This could be more aligned with the expectations of a relayer or end-user. Ex, on most exchanges I've traded on, I can't cancel an order once it's filled.

Reply:
We should either keep `cancelOrderNoThrow` or make cancels deterministic IMO. Otherwise batch cancels become unreliable. Is there any downside to allowing filled orders to be cancelled, other than gas costs?

Reply:
Also note that `getOrderInfo` will still return `FILLED` for an order that is both filled and cancelled.

Reply:
It seems like reverting if the order has `expired` would make a lot of sense since a call to `cancel` could take a few blocks to get mined, which would end up costing a lot more gas now.

Reply:
On the `FILLED` front, I think that distinction that can be handled by UX.

I think there still is a need for `batchCancelOrdersNoThrow()` because this function throws if the order is already cancelled. How about we just remove this check entirely or do nothing if the order is already cancelled?